src/lib/annotations/Annotator.js

```diff
@@ -677,23 +690,53 @@ class Annotator extends EventEmitter {
             /* eslint-enable require-jsdoc */
 
             const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
-
-            handlers.push({
-                type: 'mousemove',
-                func: annotatorUtil.eventToLocationHandler(locationFunction, drawingThread.handleMove),
-                eventObj: this.annotatedElement
-            });
-            handlers.push({
-                type: 'mousedown',
-                func: annotatorUtil.eventToLocationHandler(locationFunction, drawingThread.handleStart),
-                eventObj: this.annotatedElement
-            });
-            handlers.push({
-                type: 'mouseup',
-                func: annotatorUtil.eventToLocationHandler(locationFunction, drawingThread.handleStop),
-                eventObj: this.annotatedElement
+            const undoButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_UNDO);
+            const redoButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_REDO);
+
+            // NOTE (@minhnguyen): Move this logic to a new controller class
+            const that = this;
+            drawingThread.addListener('annotationevent', (data = {}) => {
```

Comment:
This will be done in a separate PR. I don't think it makes sense to separate and test it at the moment when it will be moved in a coming class change.